returns/tracing.py

```diff
@@ -0,0 +1,63 @@
+import types
+from contextlib import contextmanager
+from inspect import FrameInfo, stack
+from typing import List, Optional
+
+from returns.result import _Failure
+
+
+@contextmanager
+def collect_traces():
+    """
+    Context Manager/Decorator to active traces collect to the Failures.
+
+    .. code:: python
+
+        >>> from inspect import FrameInfo
+
+        >>> from returns.io import IOResult
+        >>> from returns.result import Result
+        >>> from returns.tracing import collect_traces
+
+        >>> with collect_traces():
+        ...     traced_failure = Result.from_failure('Traced Failure')
+        >>> non_traced_failure = IOResult.from_failure('Non Traced Failure')
+
+        >>> assert non_traced_failure.trace is None
+        >>> assert isinstance(traced_failure.trace, list)
+        >>> assert isinstance(traced_failure.trace[0], FrameInfo)
+        >>> traced_failure.trace  # doctest: +SKIP
+        [FrameInfo(frame=<frame at 0x7f, file '.../returns/returns/result.py', line 525, code Failure>, ...]
```

Comment:
We can prettify it somehow. Our goal it to make it readable. Possibly `pprint`?

Reply:
Maybe we can add a *for loop* with *print*!
```python
>>> for trace_line in traced_failure.trace:
...     print(f"{trace_line.filename}:{trace_line.lineno} in `{trace_line.function}`")
...
/returns/returns/result.py:529 in `Failure`
/example_folder/example.py:26 in `my_example_func`
/example_folder/example.py:53 in `<module>`
```
What do you think?

Reply:
++